### ISSUE TITLE
3.8.0b3: crash  when trying to view a smart folder

### ISSUE THREAD

**Issue by phiw13** [ViennaRSS/vienna-rss#1559]
what the headline says:
3.8.0b3: crash  when trying to view a smart folder

**To Reproduce**
* click to select a smart folder (e.g. “today” or ”all unread”).
* crash

log attached below

**Please complete the following information:**
 - Vienna version 3.8.0b3
 - OS version: 12.3.1

**Additional information:**
In which version of Vienna does the problem not occur, if applicable.

[Vienna-2022-05-16-130739.ips.zip](https://github.com/ViennaRSS/vienna-rss/files/8697863/Vienna-2022-05-16-130739.ips.zip)

**Comment by michbsd** [comment]
Seeing the same..  and not only on smart folders.

This is the crash error:

```
Application Specific Information:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString numericCompare:]: unrecognized selector sent to instance 0x6000031f0800'
abort() called
terminating with uncaught exception of type NSException
```

**Comment by phiw13** [comment]
I found one feed that is causing crashes (clicking on that one in the feed list, or its enclosing folder crashes the browser).

`https://orientxxi.info/?page=backend&lang=fr` – I have 2 unread articles for this one.

**Comment by Eitot** [comment]
I will have a look. I am seeing anything in the logs so far, but I’ll try to reproduce it with the feed above.

**Comment by sbondorf** [comment]
Worked for me, thanks!

Another detail that I now remember and that indicates I experienced this problem: after downgrading to beta 2, my articles weren't sorted ascending by date anymore.

**Comment by Eitot** [comment]
> Worked for me, thanks!
> 
> Another detail that I now remember and that indicates I experienced this problem: after downgrading to beta 2, my articles weren't sorted ascending by date anymore.

That’s unfortunately necessary, because NSArchiver/NSUnarchiver are old formats that aren’t compatible with any of their replacements that Vienna now uses (#1525). It was not possible to simply reuse the same preference for this, since it would have resulted in crashes in the older versions of Vienna.

**Comment by phiw13** [comment]
after upgrading again to beta3 I immediately quit Vienna and then tried the `defaults delete` command.

```
Domain (uk.co.opencommunity.vienna2) not found.
Defaults have not been changed.
```

That particular preference did not appear to exist :-(. I then deleted the plist – Vienna’s prefs are not hard at all to reset and restarted Vienna. Since then I have not yet experienced any of these crashes.

Like @sbondorf I had to reset the sort order after downgrading. Anyway, at the moment all seems well. Feeds and smart folders that caused crashes yesterday now display as expected.
